### ISSUE TITLE
Feature Request: Excluded path for `filter_on_submit`, `filter_as_you_type`

### ISSUE THREAD

**Issue by nyngwang** [nvim-neo-tree/neo-tree.nvim#86]
As title. This is a must-have I promise. The current situation is that in both cases some stupid folder like `~/Library/` will be scanned and cause the entire tree to get frozen & my computer very hot!

#### NVIM Version
```
NVIM v0.7.0-dev+913-ge07a4b97f
Build type: Release
LuaJIT 2.1.0-beta3
```

**Comment by cseickel** [comment]
It's a good idea, the question is how exactly to do this. 

I like the idea of excluding glob patterns, but there can be a lot of variation of what each system's built in fd/find/where command will support. Since I am only testing on my own system, I would prefer to just give people the ability to add their own arbitrary arguments to be injected into the find command we ultimately execute. That would offer the ultimate flexibility for the end user without me having to worry about supporting systems I don't have.

Because I love having everything be dynamic, I would make this option take either a string or a function. The function would be told what the root path is and what the search term is and be able to dynamically pass back extra arguments. You might want to use this to have different excludes for different root directories.

What do you think?

**Comment by nyngwang** [comment]
> [...] I would prefer to just give people the ability to add their own arbitrary arguments to be injected into the find command we ultimately execute. [...]

I was thinking about the same! Currently, you provide a string `filter_on_submit`, I was thinking about passing a lua function with some parameter:

```lua
-- old
--            ['/'] = 'open_vsplit'

-- new
            ['/'] = function (tools) -- Or we don't need this param, just use `require('neo-tree')`
              -- tools.open_vsplit()

              -- User can do other thing
            end
```

But it's hard to imagine for some cases like `filter_on_submit`, because it is the problem root.

**Comment by cseickel** [comment]
Recreating the entire command in this case would be a lot of work because it's definitely the most complicated one. I was thinking of adding this as a simplier configuration option, alongside other filter options:

```lua
require("neo-tree").setup({
  filesystem = {
    search_limit = 50,
    find_cmd = "fd",
    find_args = "--exclude SomeDir/*" --  I have no idea what the real syntax is yet
    -- or use a function instead of string
    find_args = function(path, search_term)
      if path == "/home/cseickel" then
        return "--exclude Library/*"
      else
        return "--exclude node_modules/*
      end if
    end
  }
})
```

**Comment by nyngwang** [comment]
@cseickel: I like this one. I use another plugin has a similar design and it works very well. Here is my config for reference to the structure:

```lua
  use {
    -- WARNING: fzf-lua does not include fzf. Run `brew install fzf --HEAD` first!
    'ibhagwan/fzf-lua',
    requires = {
      'kyazdani42/nvim-web-devicons'
    },
    config = function()
      local actions = require "fzf-lua.actions"
      require'fzf-lua'.setup {

--- many lines skipped
        files = {
          show_cwd_header=true,
          prompt = ' File $ ',
          -- WARNING: Run `brew install fd` to install fd.
          fd_opts =
            "--hidden "..
            "--no-ignore "..
            "--ignore-case "..
            "--type f "..
            -- "--max-depth 15 "..
            " "..
            "--exclude .local "..
            "--exclude .npm "..
            "--exclude .nvm "..
            "--exclude .tmux "..
            "--exclude .cache "..
            "--exclude .vscode "..
            " "..
            "--exclude .DS_Store "..
            "--exclude .git "..
            "--exclude node_modules "..
            "--exclude package-lock.json "..
            "--exclude yarn.lock "..
            "--exclude tags "..
            " ",
          actions = {
            -- WARNING: Required to prevent qf-list on multi-selection.
            ["default"] = actions.file_edit,
          }
        },
```

**Comment by nyngwang** [comment]
@cseickel: Is it possible to use the plain-old built-in `/` to highlight the keyword I want to search inside the file tree? While I think the problem is that I will only use floating version. Is this possible?

**Comment by cseickel** [comment]
You are supposed to be able to map "/" to "none" to go back to the built-in vim search. I'm not sure if that will work based on your other bug, but it definitely will after I fix it.

**Comment by nyngwang** [comment]
> This is in main right now

A good reason for me to switch to `main` now :)

**Comment by cseickel** [comment]
I put the wrong number in the commit. For reference, this was closed in [Release v1.23 ](https://github.com/nvim-neo-tree/neo-tree.nvim/releases/tag/v1.23)  in commit 751ee6b4e840d36915ab5ac7234e848aa45db1ac.
